sw/airborne/modules/multi/traffic_info.h

```diff
@@ -0,0 +1,383 @@
+extern void acInfoCalcVelocityEnu_f(uint8_t ac_id);
+
+/** Get position from UTM coordinates (int).
+ * @param[in] aircraft id of aircraft info to get
+ */
+static inline struct UtmCoor_i *acInfoGetPositionUtm_i(uint8_t ac_id)
+{
+  if (!bit_is_set(ti_acs[ti_acs_id[ac_id]].status, AC_INFO_POS_UTM_I)) {
+    acInfoCalcPositionUtm_i(ac_id);
+  }
+  return &ti_acs[ti_acs_id[ac_id]].utm_pos_i;
+}
+
+/** Get position from LLA coordinates (int).
+ * @param[in] aircraft id of aircraft info to get
+ */
+static inline struct LlaCoor_i *acInfoGetPositionLla_i(uint8_t ac_id)
+{
+  if (!bit_is_set(ti_acs[ti_acs_id[ac_id]].status, AC_INFO_POS_LLA_I)) {
+    acInfoCalcPositionLla_i(ac_id);
+  }
+  return &ti_acs[ti_acs_id[ac_id]].lla_pos_i;
+}
+
+/** Get position in local ENU coordinates (int).
+ * @param[in] aircraft id of aircraft info to get
+ */
+static inline struct EnuCoor_i *acInfoGetPositionEnu_i(uint8_t ac_id)
+{
+  if (!bit_is_set(ti_acs[ti_acs_id[ac_id]].status, AC_INFO_POS_ENU_I)) {
+    acInfoCalcPositionEnu_i(ac_id);
+  }
+  return &ti_acs[ti_acs_id[ac_id]].enu_pos_i;
+}
+
+/** Get position from UTM coordinates (float).
+ * @param[in] aircraft id of aircraft info to get
+ */
+static inline struct UtmCoor_f *acInfoGetPositionUtm_f(uint8_t ac_id)
+{
+  if (!bit_is_set(ti_acs[ti_acs_id[ac_id]].status, AC_INFO_POS_UTM_F)) {
+    acInfoCalcPositionUtm_f(ac_id);
+  }
+  return &ti_acs[ti_acs_id[ac_id]].utm_pos_f;
+}
+
+/** Get position from LLA coordinates (float).
+ * @param[in] aircraft id of aircraft info to get
+ */
+static inline struct LlaCoor_f *acInfoGetPositionLla_f(uint8_t ac_id)
+{
+  if (!bit_is_set(ti_acs[ti_acs_id[ac_id]].status, AC_INFO_POS_LLA_F)) {
+    acInfoCalcPositionLla_f(ac_id);
+  }
+  return &ti_acs[ti_acs_id[ac_id]].lla_pos_f;
+}
+
+/** Get position in local ENU coordinates (float).
+ * @param[in] aircraft id of aircraft info to get
+ */
+static inline struct EnuCoor_f *acInfoGetPositionEnu_f(uint8_t ac_id)
+{
+  if (!bit_is_set(ti_acs[ti_acs_id[ac_id]].status, AC_INFO_POS_ENU_F)) {
+    acInfoCalcPositionEnu_f(ac_id);
+  }
+  return &ti_acs[ti_acs_id[ac_id]].enu_pos_f;
+}
+
+/** Get position from ENU coordinates (int).
+ * @param[in] aircraft id of aircraft info to get
+ */
+static inline struct EnuCoor_i *acInfoGetVelocityEnu_i(uint8_t ac_id)
+{
+  if (!bit_is_set(ti_acs[ti_acs_id[ac_id]].status, AC_INFO_VEL_ENU_I)) {
+    acInfoCalcVelocityEnu_i(ac_id);
+  }
+  return &ti_acs[ti_acs_id[ac_id]].enu_vel_i;
+}
+
+/** Get position from ENU coordinates (float).
+ * @param[in] aircraft id of aircraft info to get
+ */
+static inline struct EnuCoor_f *acInfoGetVelocityEnu_f(uint8_t ac_id)
+{
+  if (!bit_is_set(ti_acs[ti_acs_id[ac_id]].status, AC_INFO_VEL_ENU_F)) {
+    acInfoCalcVelocityEnu_f(ac_id);
+  }
+  return &ti_acs[ti_acs_id[ac_id]].enu_vel_f;
+}
+
+/** Get vehicle course (float).
+ * @param[in] aircraft id of aircraft info to get
+ */
+static inline float *acInfoGetCourse(uint8_t ac_id)
```

Comment:
I think it would be nicer/better if all get functions returning a scalar actually return a scalar instead of a pointer (like in the state interface).